Automattic/wp-calypso
# Domains: Create new main domain info component

### Changes proposed in this Pull Request

This PR creates the new redesigned main domain info component, which is shown in the domain settings page for registered and connected domains. This is part of the domain management pages redesign project detailed in pcYYhz-m2-p2.

#### Screenshots

- New page - registered domain info:

![Markup on 2021-12-15 at 15:59:58](https://user-images.githubusercontent.com/5324818/146248450-d93fb022-5a50-4216-8e1b-38a04848b575.png)

- New page in mobile layout:

<img width="402" alt="Screen Shot 2021-12-15 at 16 01 30" src="https://user-images.githubusercontent.com/5324818/146248687-cd51d5d7-3404-45c3-bd0a-05908927db39.png">

- New page - connected domain info:

![Markup on 2021-12-15 at 20:27:48](https://user-images.githubusercontent.com/5324818/146280645-117ee018-6dd9-47f8-a46f-bb78668cbaaa.png)

- Old page:

![Markup on 2021-12-15 at 15:49:28](https://user-images.githubusercontent.com/5324818/146248207-c32ea924-7c2d-4ea5-92f1-583b6c34349c.png)

<img width="750" alt="Screen Shot 2021-12-15 at 20 25 42" src="https://user-images.githubusercontent.com/5324818/146280435-0d455e00-806e-48ea-ac71-634ba0436c56.png">

### Testing instructions

- Build this branch locally or open the live Calypso link
  - If you're in the live Calypso link, please append `?flags=domains/settings-page-redesign` to your URL to enable the appropriate feature flag
- Go to "Upgrades > Domains"
- Select one of your registered or mapped domains
- Ensure that:
  - The main info card in the domain settings page looks like the screenshot above
  - The auto-renew toggle works like in the old domain settings page
  - The renew now button works the same as in the old domain settings page
  - The "Payment details" button takes you to the domain purchase detail page
  - Also check that the mobile view looks correct
